tests/Unit/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Test_GaugeWave.cpp

```diff
@@ -0,0 +1,234 @@
+  // \alpha = \sqrt{H}
+  // \partial_t \alpha = - \partial_x H / (2 \sqrt{H})
+  // \partial_x \alpha = \partial_x H / (2 \sqrt{H})
+  // \partial_y \alpha = \partial_z \alpha = 0
+  // \beta^i = 0
+  // \partial_t \beta^i = 0
+  // \partial_i \beta^j = 0
+  // K_{ij} = 0, except K_{xx} = \partial_x H / (2 \sqrt{H})
+
+  // Parameters for GaugeWave solution
+  const double amplitude = 5.0;
+  const double wavelength = 4.0;
+  const auto x = spatial_coords(used_for_size);
+  const double t = 1.3;
+
+  // Evaluate solution
+  gr::Solutions::GaugeWave solution(amplitude, wavelength);
+
+  const auto vars =
+      solution.variables(x, t, gr::Solutions::GaugeWave::tags<DataType>{});
+  const auto& lapse = get<gr::Tags::Lapse<DataType>>(vars);
+  const auto& dt_lapse = get<Tags::dt<gr::Tags::Lapse<DataType>>>(vars);
+  const auto& d_lapse =
+      get<gr::Solutions::GaugeWave::DerivLapse<DataType>>(vars);
+  const auto& shift = get<gr::Tags::Shift<3, Frame::Inertial, DataType>>(vars);
+  const auto& d_shift =
+      get<gr::Solutions::GaugeWave::DerivShift<DataType>>(vars);
+  const auto& dt_shift =
+      get<Tags::dt<gr::Tags::Shift<3, Frame::Inertial, DataType>>>(vars);
+  const auto& g =
+      get<gr::Tags::SpatialMetric<3, Frame::Inertial, DataType>>(vars);
+  const auto& dt_g =
+      get<Tags::dt<gr::Tags::SpatialMetric<3, Frame::Inertial, DataType>>>(
+          vars);
+  const auto& d_g =
+      get<gr::Solutions::GaugeWave::DerivSpatialMetric<DataType>>(vars);
+
+  // Check those quantities that should be zero or one.
+  const auto zero = make_with_value<DataType>(x, 0.);
+  const auto one = make_with_value<DataType>(x, 1.);
+  for (size_t i = 0; i < 3; ++i) {
+    CHECK(shift.get(i) == zero);
+    CHECK(dt_shift.get(i) == zero);
+    if (i > 0) {
+      CHECK(d_lapse.get(i) == zero);
+    }
+    for (size_t j = 0; j < 3; ++j) {
+      CHECK(d_shift.get(i, j) == zero);
+      if (i != j) {
+        CHECK(g.get(i, j) == zero);
+      } else {
+        if (i > 0) {
+          CHECK(g.get(i, j) == one);
+        }
+      }
+      if (i > 0 or j > 0) {
+        CHECK(dt_g.get(i, j) == zero);
+        for (size_t k = 0; k < 3; ++k) {
+          CHECK(d_g.get(k, i, j) == zero);
+        }
+      } else {
+        for (size_t k = 1; k < 3; ++k) {
+          CHECK(d_g.get(k, i, j) == zero);
+        }
+      }
+    }
+  }
+
+  const auto& h =
+      1.0 - amplitude * sin((2.0 * M_PI / wavelength) * (get<0>(x) - t));
+  const auto& dx_h = -1.0 * (2.0 * M_PI / wavelength) * amplitude *
+                     cos((2.0 * M_PI / wavelength) * (get<0>(x) - t));
+  const auto& sqrt_h = sqrt(h);
+  const auto& dx_h_over_2_sqrt_h = 0.5 * dx_h / sqrt_h;
+
+  // Check quantities that depend on h
+  CHECK(get(lapse) == sqrt_h);
+  CHECK(get(dt_lapse) == -1.0 * dx_h_over_2_sqrt_h);
+  CHECK(get<0>(d_lapse) == dx_h_over_2_sqrt_h);
+  CHECK(get<0, 0>(g) == h);
+  CHECK(get<0, 0>(dt_g) == -1.0 * dx_h);
+  CHECK(get<0, 0, 0>(d_g) == dx_h);
```

Comment:
you use literal numbers for all of these quantities, so don't compute them in lines 128--133 but `CHECK` for literal numbers here and also in lines 151, 152.

The reason is that I trust unit tests with literal numbers way more than tests that re-implement the code they are supposed to test.

Reply:
OK, I've done this. In general, when comparing expected double literals, should I use CHECK() or CHECK_ITERABLE_APPROX?

Reply:
Actually, we want to avoid unnamed literals. This falls into the category of "magic numbers". In tests it's somewhat okay, but in general it's better to have `expected_h = ...`. The number needs to be computed somehow, which is why we typically use the python implementation; odds are the author of the test already wrote a second implementation in python/mathematica/etc so we may as well use it to do fuzzy testing.

To answer @geoffrey4444's question: for literals we typically use `CHECK(a == approx(expected_a));` unless you expect bitwise identical results, like if you're computing integers (basically anything that isn't a double)